junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/FilePosition.java

```diff
@@ -29,17 +29,21 @@
 
 	private static final long serialVersionUID = 1L;
 
-	private final int line;
-	private final int column;
-
 	/**
 	 * Create a new {@code FilePosition} using the supplied {@code line} and
 	 * {@code column}.
 	 *
 	 * @param line the line (1-based)
 	 * @param column the column (1-based)
 	 */
-	public FilePosition(int line, int column) {
+	public static FilePosition from(int line, int column) {
+		return new FilePosition(line, column);
+	}
+
+	private final int line;
+	private final int column;
+
+	private FilePosition(int line, int column) {
 		this.line = line;
 		this.column = column;
```

Comment:
@sbrannen This would probably warrant another issue, but I think that `line` and `column` may need precondition guards to ensure they are positive (> 0) numbers, as it personally makes little sense to me for line and column numbers in source files to be anything less than 1.

Reply:
What about the case where a `TestEngine` can only identify a line but not a column? Would it pass 1 as `column` then?

Reply:
@marcphilipp Good point. Perhaps we could enforce a precondition of = 0 or -1, where 0 or -1 would mean that there is no meaningful line or column? Or how about accepting `@Nullable Integer`s instead, where `null` would mean the same thing?